1.0.7: gcc 10 compile time warnings

```
sctp.c: In function 'protocol_sctp_connect':
sctp.c:196:12: warning: unused variable 'off' [-Wunused-variable]
  196 |  const int off = 0;
      |            ^~~
sctp.c:195:12: warning: unused variable 'len' [-Wunused-variable]
  195 |  socklen_t len;
      |            ^~~
udp.c: In function 'protocol_udp_listen':
udp.c:234:7: warning: unused variable 'hostname' [-Wunused-variable]
  234 |  char hostname[NI_MAXHOST];
      |       ^~~~~~~~
parse.c:333:1: warning: 'print_workorder_t' defined but not used [-Wunused-function]
  333 | print_workorder_t(workorder_t *w) {
      | ^~~~~~~~~~~~~~~~~
parse.c:280:1: warning: 'print_symbols' defined but not used [-Wunused-function]
  280 | print_symbols(struct symbol *list) {
      | ^~~~~~~~~~~~~
ssl.c: In function 'file_present':
ssl.c:133:6: warning: unused variable 'err' [-Wunused-variable]
  133 |  int err;
      |      ^~~
ssl.c: In function 'protocol_ssl_accept':
ssl.c:226:18: warning: unused variable 'name' [-Wunused-variable]
  226 |  struct sockaddr name;
      |                  ^~~~
ssl.c:224:17: warning: unused variable 'ssl_p' [-Wunused-variable]
  224 |  ssl_private_t *ssl_p = (ssl_private_t *) p->_protocol_p;
      |                 ^~~~~
ssl.c: In function 'protocol_ssl_read':
ssl.c:361:20: warning: unused variable 'flowop_options' [-Wunused-variable]
  361 |  flowop_options_t *flowop_options = (flowop_options_t *) options;
      |                    ^~~~~~~~~~~~~~
ssl.c:360:8: warning: unused variable 'host' [-Wunused-variable]
  360 |  char *host = p->host[0] == '\0' ? p->host : "Unknown";
      |        ^~~~
ssl.c:358:6: warning: unused variable 'status' [-Wunused-variable]
  358 |  int status = 1;
      |      ^~~~~~
ssl.c:357:6: warning: unused variable 'bufsize' [-Wunused-variable]
  357 |  int bufsize = size * 10;
      |      ^~~~~~~
ssl.c:356:6: warning: unused variable 'i' [-Wunused-variable]
  356 |  int i;
      |      ^
ssl.c: In function 'protocol_ssl_write':
ssl.c:377:20: warning: unused variable 'flowop_options' [-Wunused-variable]
  377 |  flowop_options_t *flowop_options = (flowop_options_t *) options;
      |                    ^~~~~~~~~~~~~~
ssl.c:376:6: warning: unused variable 'bufsize' [-Wunused-variable]
  376 |  int bufsize = size * 10;
      |      ^~~~~~~
ssl.c:375:6: warning: unused variable 'i' [-Wunused-variable]
  375 |  int i;
      |      ^
ssl.c:374:6: warning: unused variable 'status' [-Wunused-variable]
  374 |  int status = 1;
      |      ^~~~~~
ssl.c:373:8: warning: unused variable 'host' [-Wunused-variable]
  373 |  char *host = p->host[0] == '\0' ? p->host : "Unknown";
      |        ^~~~
ssl.c: In function 'ssl_fini':
ssl.c:506:17: warning: unused variable 'ssl_p' [-Wunused-variable]
  506 |  ssl_private_t *ssl_p = (ssl_private_t *) p->_protocol_p;
      |                 ^~~~~
At top level:
ssl.c:94:1: warning: 'id_function' defined but not used [-Wunused-function]
   94 | id_function(void)
      | ^~~~~~~~~~~
ssl.c:84:1: warning: 'locking_function' defined but not used [-Wunused-function]
   84 | locking_function(int mode, int n, char *file, int line)
      | ^~~~~~~~~~~~~~~~
```

--- a/src/netstat.c
+++ b/src/netstat.c
@@ -337,7 +337,6 @@ netstat_init()
 	char buffer[1024];
 	char *token;
 	int index;
-	struct stat buf;
 
 	if ((f = popen(NETSTAT_HDR, "r")) == NULL) {
 		printf("Cannot open %s\n", NETSTAT_HDR);
@@ -426,8 +425,7 @@ netstat_snap(int snaptype)
 {
 	FILE *f;
 	char buffer[1024], hdr[1024];
-	char *interface, *tmp;
-	struct packet_stats ps;
+	char *interface;
 
 	if ((f = popen(NETSTAT_DEV, "r")) == NULL) {
 		printf("Cannot open %s\n", NETSTAT_DEV);

--- a/src/parse.c
+++ b/src/parse.c
@@ -276,6 +276,8 @@ parse(char *buffer, struct symbol *list)
 	return (0);
 }
 
+#ifdef DEBUG
+
 static void
 print_symbols(struct symbol *list) {
 	while (list) {
@@ -283,6 +285,7 @@ print_symbols(struct symbol *list) {
 		list = list->next;
 	}
 }
+
 void
 print_txn_t(txn_t *t)
 {
@@ -335,6 +338,8 @@ print_workorder_t(workorder_t *w) {
 	for (i = 0; i < w->ngrp; i++)
 		print_group_t_recurse(&w->grp[i]);
 }
+
+#endif /* DEBUG */
 
 static int
 string2int(char *value)
@@ -978,7 +983,9 @@ parse_app_profile(char *filename)
 	buffer[size - 1] = '\0';
 
 	parse(buffer, &list);
-	/* print_symbols(list.next); */
+#ifdef DEBUG
+	print_symbols(list.next); */
+#endif /* DEBUG */
 	w = build_worklist(list.next);
 	if (w == NULL || no_errs > 0) {
 		print_errors();

--- a/src/parse.c
+++ b/src/parse.c
@@ -276,7 +276,7 @@ parse(char *buffer, struct symbol *list)
 	return (0);
 }
 
-#ifdef DEBUG
+#if defined(DEBUG) || defined(TEST_PARSE)
 
 static void
 print_symbols(struct symbol *list) {

--- a/src/parse.c
+++ b/src/parse.c
@@ -984,7 +984,7 @@ parse_app_profile(char *filename)
 
 	parse(buffer, &list);
 #ifdef DEBUG
-	print_symbols(list.next); */
+	print_symbols(list.next);
 #endif /* DEBUG */
 	w = build_worklist(list.next);
 	if (w == NULL || no_errs > 0) {

--- a/src/sctp.c
+++ b/src/sctp.c
@@ -192,8 +192,6 @@ protocol_sctp_connect(protocol_t *p, void *options)
 #endif
 	struct sockaddr_storage serv;
 	flowop_options_t *flowop_options = (flowop_options_t *)options;
-	socklen_t len;
-	const int off = 0;
 	char msg[128];
 
 	uperf_debug("sctp: Connecting to %s:%d\n", p->host, p->port);

--- a/src/ssl.c
+++ b/src/ssl.c
@@ -357,7 +357,7 @@ protocol_ssl_read(protocol_t * p, void *buffer, int size, void *options)
 	ssl_private_t *ssl_p = (ssl_private_t *) p->_protocol_p;
 
 	uperf_debug("ssl - Reading %d bytes from %s:%d\n", size,
-            p->host[0] == '\0' ? p->host : "Unknown",
+            p->host[0] == '\0' ? "Unknown" : p->host,
             p->port);
 
 	return (ssl_read(ssl_p->ssl, buffer, size));
@@ -367,9 +367,9 @@ static int
 protocol_ssl_write(protocol_t * p, void *buffer, int size, void *options)
 {
 	ssl_private_t *ssl_p = (ssl_private_t *) p->_protocol_p;
-	char *host = p->host[0] == '\0' ? p->host : "Unknown";
 
-	uperf_debug("ssl - Writing %d bytes to %s:%d\n", size, host,
+	uperf_debug("ssl - Writing %d bytes to %s:%d\n", size,
+            p->host[0] == '\0' ? "Unknown" : p->host,
 		p->port);
 	return (ssl_write(ssl_p->ssl, buffer, size));
 }

--- a/src/ssl.c
+++ b/src/ssl.c
@@ -62,7 +62,7 @@ typedef struct ssl_private {
 
 extern options_t options;
 
-static protocol_t * protocol_ssl_new();
+static protocol_t *protocol_ssl_new();
 /* Utility functions */
 static SSL_CTX *initialize_ctx(char *keyfile, char *password,
 	const char *method);
@@ -95,7 +95,7 @@ id_function(void)
 }
 
 static int
-my_ssl_error(SSL * ssl, int sz)
+my_ssl_error(SSL *ssl, int sz)
 {
 	char error[128];
 	int ret;
@@ -183,7 +183,7 @@ ssl_init(void *arg)
 }
 
 static ssize_t
-ssl_read(SSL * ssl, char *ptr, size_t n)
+ssl_read(SSL *ssl, char *ptr, size_t n)
 {
 	ssize_t nread;
 
@@ -192,7 +192,7 @@ ssl_read(SSL * ssl, char *ptr, size_t n)
 }
 
 static ssize_t
-ssl_write(SSL * ssl, char *ptr, size_t n)
+ssl_write(SSL *ssl, char *ptr, size_t n)
 {
 	ssize_t nwritten;
 
@@ -201,7 +201,7 @@ ssl_write(SSL * ssl, char *ptr, size_t n)
 }
 
 static int
-protocol_ssl_listen(protocol_t * p, void *o)
+protocol_ssl_listen(protocol_t *p, void *o)
 {
 	char msg[128];
 
@@ -218,7 +218,7 @@ protocol_ssl_listen(protocol_t * p, void *o)
 }
 
 static protocol_t *
-protocol_ssl_accept(protocol_t * p, void *options)
+protocol_ssl_accept(protocol_t *p, void *options)
 {
 	protocol_t *newp;
 	struct sockaddr_in remote;
@@ -269,7 +269,7 @@ protocol_ssl_accept(protocol_t * p, void *options)
 }
 
 static int
-protocol_ssl_connect(protocol_t * p, void *options)
+protocol_ssl_connect(protocol_t *p, void *options)
 {
 	struct sockaddr_storage serv;
 	BIO *sbio;
@@ -315,7 +315,7 @@ protocol_ssl_connect(protocol_t * p, void *options)
 }
 
 static int
-protocol_ssl_disconnect(protocol_t * p)
+protocol_ssl_disconnect(protocol_t *p)
 {
 	int r;
 	ssl_private_t *ssl_p = (ssl_private_t *) p->_protocol_p;
@@ -352,7 +352,7 @@ protocol_ssl_disconnect(protocol_t * p)
 }
 
 static int
-protocol_ssl_read(protocol_t * p, void *buffer, int size, void *options)
+protocol_ssl_read(protocol_t *p, void *buffer, int size, void *options)
 {
 	ssl_private_t *ssl_p = (ssl_private_t *) p->_protocol_p;
 
@@ -364,7 +364,7 @@ protocol_ssl_read(protocol_t * p, void *buffer, int size, void *options)
 }
 
 static int
-protocol_ssl_write(protocol_t * p, void *buffer, int size, void *options)
+protocol_ssl_write(protocol_t *p, void *buffer, int size, void *options)
 {
 	ssl_private_t *ssl_p = (ssl_private_t *) p->_protocol_p;
 
@@ -375,7 +375,7 @@ protocol_ssl_write(protocol_t * p, void *buffer, int size, void *options)
 }
 
 static int
-protocol_ssl_undefined(protocol_t * p, void *options)
+protocol_ssl_undefined(protocol_t *p, void *options)
 {
 	uperf_error("Undefined function in protocol called\n");
 	return (-1);
@@ -494,7 +494,7 @@ protocol_ssl_create(char *host, int port)
 }
 
 void
-ssl_fini(protocol_t * p)
+ssl_fini(protocol_t *p)
 {
 	if (!p) {
 		return;

--- a/src/ssl.c
+++ b/src/ssl.c
@@ -128,7 +128,6 @@ static int
 file_present(char *file)
 {
 	struct stat rbuf;
-
 	if (stat(file, &rbuf) != 0) {
 		return (-1);
 	}
@@ -225,9 +224,7 @@ protocol_ssl_accept(protocol_t * p, void *options)
 	struct sockaddr_in remote;
 	socklen_t addrlen;
 	int ret;
-	ssl_private_t *ssl_p = (ssl_private_t *) p->_protocol_p;
 	ssl_private_t *new_ssl_p;
-	struct sockaddr name;
 	char hostname[128];
 	flowop_options_t *flowop_options = (flowop_options_t *) options;
 	BIO *sbio;
@@ -357,15 +354,11 @@ protocol_ssl_disconnect(protocol_t * p)
 static int
 protocol_ssl_read(protocol_t * p, void *buffer, int size, void *options)
 {
-	int i;
-	int bufsize = size * 10;
-	int status = 1;
 	ssl_private_t *ssl_p = (ssl_private_t *) p->_protocol_p;
-	char *host = p->host[0] == '\0' ? p->host : "Unknown";
-	flowop_options_t *flowop_options = (flowop_options_t *) options;
 
-	uperf_debug("ssl - Reading %d bytes from %s:%d\n", size, host,
-		p->port);
+	uperf_debug("ssl - Reading %d bytes from %s:%d\n", size,
+            p->host[0] == '\0' ? p->host : "Unknown",
+            p->port);
 
 	return (ssl_read(ssl_p->ssl, buffer, size));
 }
@@ -375,10 +368,6 @@ protocol_ssl_write(protocol_t * p, void *buffer, int size, void *options)
 {
 	ssl_private_t *ssl_p = (ssl_private_t *) p->_protocol_p;
 	char *host = p->host[0] == '\0' ? p->host : "Unknown";
-	int status = 1;
-	int i;
-	int bufsize = size * 10;
-	flowop_options_t *flowop_options = (flowop_options_t *) options;
 
 	uperf_debug("ssl - Writing %d bytes to %s:%d\n", size, host,
 		p->port);
@@ -507,9 +496,6 @@ protocol_ssl_create(char *host, int port)
 void
 ssl_fini(protocol_t * p)
 {
-	ssl_private_t *ssl_p = (ssl_private_t *) p->_protocol_p;
-
-	/* free(ssl_p); */
 	if (!p) {
 		return;
 	}

--- a/src/udp.c
+++ b/src/udp.c
@@ -107,7 +107,7 @@ static int
 write_one(int fd, char *buffer, int len, struct sockaddr *to)
 {
 	socklen_t length;
-	
+
 	switch (to->sa_family) {
 	case AF_INET:
 		length = (socklen_t)sizeof(struct sockaddr_in);
@@ -231,9 +231,8 @@ protocol_udp_listen(protocol_t *p, void *options)
 {
 	udp_private_data *pd = (udp_private_data *)p->_protocol_p;
 	socklen_t len;
-	char hostname[NI_MAXHOST];
 	char msg[128];
-	
+
 	if ((pd->sock = socket(AF_INET6, SOCK_DGRAM, 0)) < 0) {
 		if ((pd->sock = socket(AF_INET, SOCK_DGRAM, 0)) < 0) {
 			(void) snprintf(msg, 128, "%s: Cannot create socket", "udp");

